app/src/ui/create-branch/create-branch-dialog.tsx

```diff
@@ -28,11 +28,16 @@ import { CommitOneLine } from '../../models/commit'
 import { PopupType } from '../../models/popup'
 import { RepositorySettingsTab } from '../repository-settings/repository-settings'
 import { isRepositoryWithForkedGitHubRepository } from '../../models/repository'
+import { debounce } from 'lodash'
+import { API } from '../../lib/api'
+import { Account } from '../../models/account'
+import { parseRepoRules } from '../../lib/helpers/repo-rules'
 
 interface ICreateBranchProps {
   readonly repository: Repository
   readonly targetCommit?: CommitOneLine
   readonly upstreamGitHubRepository: GitHubRepository | null
+  readonly gitHubAccount: Account | null
```

Comment:
I imagine these branch rules will eventually make it to GHES as well right? So we should probably pass in all accounts here and use `getAccountForRepository` to resolve the proper account for the `upstreamGitHubRepository`. We should also add a method to `endpoint-capabilities.ts` that lets us query whether the endpoint supports repo rules at runtime

Reply:
Added this in https://github.com/desktop/desktop/pull/16904/commits/d9e3c23a2bcefc6d1abcc96cbc3420b80e5325b2